clean_project/search.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+"""
+Search a word in a file and return all the occurrence specified the line numbers
+"""
+import argparse
+import json
+
+BOLD = '\033[1m'
+END = '\033[0m'
+
+
+def search(search_word, json_path):
+    """
+    Print all the occurrence of a given word
+    :param search_word: Word given to search
+    :param json_path: Indexing file
+    :return: List of files and numbers of file when search_word occurrence
+    """
+    with open(json_path) as json_file:
+        list_of_dict = json.load(json_file)
+        count_match = 0
+        result_list = []
+        for file_dict in list_of_dict:
```

Comment:
i wonder whether you can avoid iterating all dicts instead... imagine every file has one line only and there are millions of files. if the number of files is N, this implementation would be o(N) (linear) in time because you'd have to visit every single dict.

can you imagine another way to store and search more efficiently, maybe?